File: algorithm/JavaScript/ValidTriangleNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var triangleNumber = function(nums) {
    const n = nums.length;
    nums.sort((a, b) => a - b)
    let res = 0

    for (let i = 0; i < n; i++) {
      let k = i
      for (let j = i+1; j < n; j++) {
        while (k + 1 < n && nums[k+1] < nums[i] + nums[j]) {
          k += 1
        }
        res += Math.max(k - j , 0)
      }
    }

    return res

};

File: algorithm/python/ValidTriangleNumber.py
class Solution:
    def triangleNumber(self, nums):
        n = len(nums)
        res = 0
        nums.sort()
        for i in range(n):
            k = i
            for j in range(i+1, n):
                while k + 1 < n and nums[k + 1] < nums[i] + nums[j]:
                    k += 1
                res += max(k - j , 0)
        return res